File: app/src/main/java/com/esbati/keivan/persiancalendar/Services/NotificationActionService.java
package com.esbati.keivan.persiancalendar.Services;

import android.app.IntentService;
import android.app.NotificationManager;
import android.content.Context;
import android.content.Intent;
import android.support.annotation.Nullable;

import com.esbati.keivan.persiancalendar.Utils.NotificationHelper;

/**
 * Created by Keivan Esbati on 4/9/2017.
 */

public class NotificationActionService extends IntentService {

    public final static String EXTRA_ACTION = "extra_action";
    public final static int ACTION_DISMISS = 0;

    public NotificationActionService() {
        super("NotificationActionService");
    }

    @Override
    protected void onHandleIntent(@Nullable Intent intent) {
        int mActionType = intent.getIntExtra(EXTRA_ACTION, -1);

        if(mActionType >= 0)
            switch (mActionType){
                case 0:
                    //Dismiss Notification
                    NotificationManager mNotificationManager =
                            (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
                    mNotificationManager.cancel(NotificationHelper.NOTIFICATION_ID);
                    break;
            }
    }
}

File: app/src/main/java/com/esbati/keivan/persiancalendar/features/notification/NotificationActionService.kt
package com.esbati.keivan.persiancalendar.features.notification

import android.app.IntentService
import android.app.PendingIntent
import android.content.Context
import android.content.Intent

/**
 * Created by Keivan Esbati on 4/9/2017.
 */

private const val EXTRA_ACTION = "extra_action"
private const val ACTION_DISMISS = 0

class NotificationActionService : IntentService("NotificationActionService") {

    override fun onHandleIntent(intent: Intent?) {
        val mActionType = intent!!.getIntExtra(EXTRA_ACTION, -1)

        if (mActionType >= 0)
            when (mActionType) {
                ACTION_DISMISS -> NotificationHelper.cancelNotification(this)
            }
    }

    companion object {
        fun getDismissAction(context: Context): PendingIntent {
            val dismissIntent = Intent(context, NotificationActionService::class.java).apply {
                putExtra(EXTRA_ACTION, ACTION_DISMISS)
            }
            return PendingIntent.getService(context, 0, dismissIntent, PendingIntent.FLAG_CANCEL_CURRENT)
        }
    }
}
